jetbrains-core/src/software/aws/toolkits/jetbrains/ui/connection/AwsConnectionSettingsSelector.kt

```diff
@@ -16,14 +16,20 @@ import javax.swing.event.PopupMenuEvent
 
 class AwsConnectionSettingsSelector(
     project: Project?,
-    private val settingsChangedListener: (ConnectionSettings?) -> Unit = { _ -> }
+    serviceId: String? = null,
+    private val settingsChangedListener: (ConnectionSettings?) -> Unit = { _ -> },
 ) {
     private val regionProvider = AwsRegionProvider.getInstance()
     private val credentialManager = CredentialManager.getInstance()
     val view = AwsConnectionSettings()
 
     init {
-        view.region.setRegions(regionProvider.allRegions().values.toMutableList())
+        val regions = if (serviceId != null) {
+            regionProvider.allRegionsForService(serviceId).values.toMutableList()
```

Comment:
is it possible to have invalid regions for a service show up, but not select-able (and maybe put them at the bottom of the list or something)?

Reply:
I think that requires a custom renderer but it's probably possible, Why would we want to do that though?